src/test/rustdoc-ui/deny-intra-link-resolution-failure.stderr

```diff
@@ -2,13 +2,14 @@ error: unresolved link to `v2`
   --> $DIR/deny-intra-link-resolution-failure.rs:3:6
    |
 LL | /// [v2]
-   |      ^^ unresolved link
+   |      ^^
    |
 note: the lint level is defined here
   --> $DIR/deny-intra-link-resolution-failure.rs:1:9
    |
 LL | #![deny(broken_intra_doc_links)]
    |         ^^^^^^^^^^^^^^^^^^^^^^
+   = note: no item named `v2` is in scope
```

Comment:
Make this a `span_label` on the main span. We probably want to provide some extra context about what the current scope is, but that can be done in a separate PR.

Reply:
> We probably want to provide some extra context about what the current scope is, but that can be done in a separate PR

This ended up being really easy to do so I added it here. Let me know if I should separate it out.